prelude/prelude.purs

```diff
@@ -702,6 +706,167 @@ module Prelude
       EQ -> compare xs ys
       other -> other
 
+  -- | The `Bounded` type class represents types that are finite partially
+  -- | ordered sets.
+  -- |
+  -- | Instances should satisfy the following law in addition to the `Ord` laws:
+  -- |
+  -- | - Ordering: `bottom <= a <= top`
+  class (Ord a) <= Bounded a where
+    top :: a
+    bottom :: a
+
+  instance boundedBoolean :: Bounded Boolean where
+    top = true
+    bottom = false
+
+  -- | The `Lattice` type class represents types that are partially ordered
+  -- | sets with a supremum (`sup` or `||`) and infimum (`inf` or `&&`).
+  -- |
+  -- | Instances should satisfy the following laws in addition to the `Ord`
+  -- | laws:
+  -- |
```

Comment:
Yeah, I wasn't sure which way to go with that, but I figured it would be easier to read with the operators. That's also why I listed their equivalents in the comment above, even though they're not part of the class definition itself.